mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -247,4 +247,21 @@ object MLTestingUtils extends SparkFunSuite {
     }
     models.sliding(2).foreach { case Seq(m1, m2) => modelEquals(m1, m2)}
   }
+
+  /**
+   * Helper function for testing different input types for features. Given a DataFrame, generate
+   * three output DataFrames: one having vector feature column with float precision, one having
+   * double array feature column with float precision, and one having float array feature column.
+   */
+  def generateArrayFeatureDataset(dataset: Dataset[_]): (Dataset[_], Dataset[_], Dataset[_]) = {
+    val toFloatVectorUDF = udf { (features: Vector) => features.toArray.map(_.toFloat).toVector}
+    val toDoubleArrayUDF = udf { (features: Vector) => features.toArray}
+    val toFloatArrayUDF = udf { (features: Vector) => features.toArray.map(_.toFloat)}
+    val newDataset = dataset.withColumn("features", toFloatVectorUDF(col("features")))
```

Comment:
minor: maybe useful to define `"features"` as a constant at the beginning of the function